packages/ifpack2/test/unit_tests/Ifpack2_UnitTestOverlappingRowMatrix.cpp

```diff
@@ -109,6 +110,153 @@ typedef Tpetra::global_size_t GST;
 } while (false)
 
 
+
+
+/***********************************************************************************/
+template<class MatrixClass, class MultiVectorClass>
+void localReducedMatvec(const MatrixClass & A_lcl,
+                        const MultiVectorClass & X_lcl,
+                        const int userNumRows,
+                        MultiVectorClass & Y_lcl) {
+  using Teuchos::NO_TRANS;
+
+  using execution_space = typename MatrixClass::execution_space;
+
+  if (A_lcl.numRows() == 0 || userNumRows ==0 || userNumRows > A_lcl.numRows()) {
+    return;
+  }
+
+  int team_size = -1;
+  int vector_length = -1;
+  int64_t rows_per_thread = -1;
+  
+  int64_t numLocalRows = userNumRows;
+  int64_t myNnz = A_lcl.nnz();
+
+  int64_t rows_per_team = 
+    Tpetra::Details::residual_launch_parameters<execution_space>(numLocalRows, myNnz, rows_per_thread, team_size, vector_length);
+  int64_t worksets = (X_lcl.extent (0) + rows_per_team - 1) / rows_per_team;
+
+  using policy_type = typename Kokkos::TeamPolicy<execution_space>;
```

Comment:
If you use `int64_t` for `worksets`, then you should also use `int64_t` as the index type for `Kokkos::TeamPolicy`.  (I doubt you need `int64_t`, though, since you're limited by the type of `A_lcl.numRows()` anyway.)